livekit_models.proto

```diff
@@ -391,8 +392,124 @@ enum ReconnectReason {
   RR_SWITCH_CANDIDATE = 4;
 }
 
+enum SubscriptionStatus {
+  Unsubscribed = 0;
+  Desired = 1;
+  Subscribed = 2;
+}
+
 enum SubscriptionError {
   SE_UNKNOWN = 0;
   SE_CODEC_UNSUPPORTED = 1;
   SE_TRACK_NOTFOUND = 2;
 }
+
+message ClientReport {
+
+  message RTCSenderStats {
+    // ice-candidate-pair
+    uint32 available_outgoing_bitrate = 1;
+    // local-candidate
+    string networkType = 3;
+    // outbound-rtp
+    uint64 packetsSent = 4;
+    uint32 packetsLost = 5;
+    uint32 nack_count = 6;    
+    uint32 retransmitted_packets_sent = 7;
+    uint64 timestamp = 8;
+    string stream_id = 9;
+  }
+
+  message RTCAudioSenderStats {
+   RTCSenderStats common_stats = 1;
+   // media-source
+   double total_audio_energy = 2;
+   double total_samples_duration = 3;
+  }
+
+  message RTCVideoSenderStats {
+   RTCSenderStats common_stats = 1;
+    // outbound-rtp
+    map<string, uint32> quality_limitation_durations = 2;
+    string quality_limitation_reason = 3;
+    uint32 quality_limitation_resolution_changes = 4;
+    string scalability_mode = 5;
+    uint32 frame_height = 6;
```

Comment:
which layer would dimension stats be referencing? I think we also already have these pieces of data on the server side.

Reply:
the original dimension. Makes sense to omit them, if we have them already.